src/views/clusteroverview/overview/components/top-consumers-card/utils/TopConsumerCard.tsx

```diff
@@ -52,20 +52,20 @@ const TopConsumerCard: React.FC<TopConsumersMetricCard> = ({ cardID }) => {
             isCheckboxSelectionBadgeHidden
           >
             {TopConsumerMetric.getAll().map((metric) => (
-              <SelectOption key={metric.getValue()} value={t(metric.getDropdownLabel())} />
+              <SelectOption key={metric.getValue()} value={t(metric?.getDropdownLabel())} />
             ))}
           </FormPFSelect>
         </div>
         <div className="kv-top-consumer-card__scope-select">
           <FormPFSelect
             toggleId="kv-top-consumers-card-scope-select"
             variant={SelectVariant.single}
-            selections={t(TopConsumerScope.fromString(scopeKey).getDropdownLabel())}
+            selections={t(TopConsumerScope.fromString(scopeKey)?.getDropdownLabel())}
```

Comment:
is TopConsumerScope guarenteed to be defined? if not add here ? as well

Reply:
yes. It's a class and fromString is a static method of this class